### PR TITLE
Task/sapnamysore/tlt 2645/migrate site creator

### PR DESCRIPTION
**Pull request by sapnamysore**
This PR contains the changes to move the Canvas site creator from icommons_lti_tools to admin console. 

_Note:  1. The triangular dependency has also been removed (dependency on icommons_ext_tools to run the jobs in addition to importing dccsw). dccsw is being imported but the asynchronous jobs no longer need to be  invoked from icommons_ext_tools  - and all the email settings have been moved to this repository and tested on dev server._

2. The selenium changes are in a separate PR based on this branch. 

Deployed and tested the code on dev: https://canvas.dev.tlt.harvard.edu/accounts/8/external_tools/44

### CODE REVIEW

File: canvas_site_creator/api.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        int(request.GET.get('start', 0)),
+        int(request.GET.get('length', 10)),
+        int(request.GET.get('order[0][column]', 3)),
+        request.GET.get('order[0][dir]', 'asc'),
+        request.GET.get('search[value]')
+    )
+
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
```
Comment (elliottyates) [inline canvas_site_creator/api.py]: Do we want to make these permissions a little more local/based on what's provided in this project's `settings/base.py`?

File: canvas_site_creator/templates/canvas_site_creator/base.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <script src="{% static 'angular-1.5.2/angular.min.js' %}"></script>
+    <script src="{% static 'angular-1.5.2/angular-route.min.js' %}"></script>
+    <script src="{% static 'angular-1.5.2/angular-sanitize.min.js' %}"></script>
+    <script src="{% static 'angular-1.5.2/angular-animate.min.js' %}"></script>
+    <script src="{% static 'angular-ui-bootstrap-0.14.3/ui-bootstrap-tpls-0.14.3.min.js' %}"></script>
+    <script src="{% static 'djangular/js/django-angular.min.js' %}"></script>
+{% endif %}
+    {# fixes safari autocomplete issue #}
+    {# (see https://github.com/angular/angular.js/issues/1460) #}
+    <script src="{% static 'js/polyfill/autofill-event.js' %}"></script>
```
Comment (elliottyates) [inline canvas_site_creator/templates/canvas_site_creator/base.html]: This doesn't resolve; we could use one packaged directly in this project, e.g. 
```
    <script src="{% static 'course_info/js/polyfills/autofill-event.js' %}"></script>
```

File: canvas_site_creator/templates/canvas_site_creator/create_new_course.html [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    <link rel="stylesheet" type="text/css" href="{% static 'bootstrap-3.3.6/dist/css/bootstrap.css' %}" media="screen"/>
+    <link rel="stylesheet" type="text/css" href="{% static 'datatables-1.10.7/media/css/jquery.dataTables.css' %}" media="screen"/>
+  {% else %}
+    <link rel="stylesheet" type="text/css" href="{% static 'bootstrap-3.3.6/dist/css/bootstrap.min.css' %}" media="screen"/>
+    <link rel="stylesheet" type="text/css" href="{% static 'datatables-1.10.7/media/css/jquery.dataTables.min.css' %}" media="screen"/>
+  {% endif %}
+{% endblock css %}
+
+{% block js %}
+  {% if debug %}
```
Comment (elliottyates) [inline canvas_site_creator/templates/canvas_site_creator/create_new_course.html]: These two don't resolve; we need to specify the `dist` parent dir like so:
```
  {% if debug %}
    <script src="{% static 'bootstrap-3.3.6/dist/js/bootstrap.js' %}"></script>
  {% else %}
    <script src="{% static 'bootstrap-3.3.6/dist/js/bootstrap.min.js' %}"></script>
  {% endif %}
```

**Comment by elliottyates** [comment]
Great work! It ran as expected for me locally. Excited to see this move! Have we raised tech debt to remove it from icommons_lti_tools as well? In any case, made some recommendations, and otherwise 👍 , thanks!

File: canvas_site_creator/templates/canvas_site_creator/create_new_course.html (hunk 3 above)
Comment (sapnamysore) [inline canvas_site_creator/templates/canvas_site_creator/create_new_course.html]: yes, thank u! Correctly specified in restricted_access.html, but missed it here..

**Comment by sapnamysore** [comment]
There is a  tech debt to handle the cleanup of code from icommons_lti_tool and ex_tools : https://jira.huit.harvard.edu/browse/TLT-2920

**Pull request merged**